crates/otel/src/metric.rs

```diff
@@ -0,0 +1,37 @@
+//! Global metrics registry.
+//!
+//! ## Convention for metrics in godwoken:
+//!
+//! Each crate/module/component can define their own metrics and register them
+//! to the global `REGISTRY`. To avoid naming conflict, each of them SHOULD use
+//! a unique prefix, e.g. the crate name or component name.
+//!
+//! If it makes sense to define a metrics struct, e.g. when there are a few
+//! related metrics and they usually change together, it SHOULD live in a
+//! separate metrics module. See the metrics module in gw-chain for an example.
+//!
+//! When you add/modify some metrics, make sure to update the metrics document
+//! in docs/metrics.md.
+//!
+use std::sync::{RwLock, RwLockWriteGuard};
+
+pub use once_cell::sync::{Lazy, OnceCell};
+pub use prometheus_client;
```

Comment:
Why is this module in `otel`? It could be a bit confusing because opentelemetry metrics and openmetrics (prometheus) are two different metrics solution, and we are actually using the latter, not opentelemetry metrics.

Reply:
Any suggestion? I want them in one crate. I also consider switching to opentelemetry metric in the future.

Reply:
Why should they be in one crate?

Reply:
Original `gw-metrics` merely expose `REGISTRY`, i think we can merge them into one.

Reply:
I think we can rename this crate to `gw-telemetry`. It's more general than otel (opentelemetry) and also easier to understand than an abbreviation.

Reply:
fixed